File: src/main/java/de/steklopod/sec10/Lec01Repeat.java
package de.steklopod.sec10;

import de.steklopod.utils.Util;
import reactor.core.publisher.Flux;

import java.util.concurrent.atomic.AtomicInteger;

public class Lec01Repeat {

    private static AtomicInteger atomicInteger = new AtomicInteger(1);

    public static void main(String[] args) {

        getIntegers()
                .repeat(() -> atomicInteger.get() < 14)
                .subscribe(Util.subscriber());


    }


    private static Flux<Integer> getIntegers(){
        return Flux.range(1, 3)
                    .doOnSubscribe(s -> System.out.println("Subscribed"))
                    .doOnComplete(() -> System.out.println("--Completed"))
                    .map(i -> atomicInteger.getAndIncrement());
    }

}

File: src/main/kotlin/de/steklopod/sec10/Lec01Repeat.kt
package de.steklopod.sec10

import de.steklopod.utils.Util.subscriber
import reactor.core.publisher.Flux
import java.util.concurrent.atomic.AtomicInteger

object Lec01Repeat {
    private val atomicInteger = AtomicInteger(1)

    @JvmStatic
    fun main(args: Array<String>) {
        integers()
            .repeat { atomicInteger.get() < 14 }
            .subscribe(subscriber())
    }

    private fun integers(): Flux<Int> = Flux.range(1, 3)
            .doOnSubscribe { println("Subscribed") }
            .doOnComplete { println("--Completed") }
            .map { atomicInteger.getAndIncrement() }
}
